ciao-controller/instance.go

```diff
@@ -46,7 +46,7 @@ type instance struct {
 	startTime time.Time
 }
 
-func isCNCIWorkload(workload *types.Workload) bool {
+func isCNCIWorkload(workload types.Workload) bool {
```

Comment:
I'm not sure the compiler will avoid the copy here.  The spec says the the function will receive a copy of the argument and doesn't mention that optimizations are allowed.  I guess we could check the assembly.  But in general I think it's reasonable to pass values by type if the parameter is large.  In this case it's 160 bytes.  The best justification I can find is here:

https://github.com/golang/go/wiki/CodeReviewComments#receiver-type

The 5th bullet point states that

```If the receiver is a large struct or array, a pointer receiver is more efficient. How large is large? Assume it's equivalent to passing all its elements as arguments to the method. If that feels too large, it's also too large for the receiver.
```

A receiver here is just an argument so I'm extrapolating to arguments in general.  The fact that it's faster to use pointer rather than value receivers implies that for receivers at least the compiler doesn't do any optimisation.